nativelib/src/main/scala/scala/scalanative/unsafe/Word.scala

```diff
@@ -0,0 +1,353 @@
+package scala.scalanative
+package unsafe
+
+import scala.runtime.BoxesRunTime._
+import scala.reflect.ClassTag
+
+import scalanative.annotation.alwaysinline
+
+import scalanative.runtime._
+import scalanative.runtime.Intrinsics._
+import scalanative.runtime.Boxes._
+
+import scalanative.unsigned._
+
+final class Word(private[scalanative] val rawWord: RawWord) {
+  @alwaysinline def toByte: Byte   = castRawWordToInt(rawWord).toByte
+  @alwaysinline def toChar: Char   = castRawWordToInt(rawWord).toChar
+  @alwaysinline def toShort: Short = castRawWordToInt(rawWord).toShort
+  @alwaysinline def toInt: Int     = castRawWordToInt(rawWord)
+  @alwaysinline def toLong: Long   = castRawWordToLong(rawWord)
+
+  @alwaysinline def toUByte: UByte   = toByte.toUByte
+  @alwaysinline def toUShort: UShort = toShort.toUShort
+  @alwaysinline def toUInt: UInt     = toInt.toUInt
+  @alwaysinline def toULong: ULong   = toLong.toULong
+  @alwaysinline def toUWord: UWord   = new UWord(rawWord)
+
+  override def hashCode: Int = java.lang.Long.hashCode(toLong)
```

Comment:
Newline after `=`.

Reply:
This also has problem wrt cooperative equality but lets keep it this way for now.